Mage.Sets/src/mage/cards/b/BarrowBlade.java

```diff
@@ -0,0 +1,105 @@
+package mage.cards.b;
+
+import java.util.UUID;
+
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.LoseAllAbilitiesTargetEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.constants.*;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author TiagoMDG
+ */
+public final class BarrowBlade extends CardImpl {
+
+    public BarrowBlade(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{1}");
+        
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // Equipped creature gets +1/+1.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new BoostEquippedEffect(1, 1)));
+
+        // Whenever equipped creature blocks or becomes blocked by a creature, that creature loses all abilities until end of turn.
+        this.addAbility(new BarrowBladeBlocksOrBecomesBlockedTriggeredAbility());
+
+
+        // Equip {1}
+        this.addAbility(new EquipAbility(Outcome.BoostCreature, new GenericManaCost(1), new TargetControlledCreaturePermanent(), true));
+    }
+
+    private BarrowBlade(final BarrowBlade card) {
+        super(card);
+    }
+
+    @Override
+    public BarrowBlade copy() {
+        return new BarrowBlade(this);
+    }
+}
+
+class BarrowBladeBlocksOrBecomesBlockedTriggeredAbility extends TriggeredAbilityImpl {
```

Comment:
this can be much simpler, see existing `BlocksOrBlockedAttachedTriggeredAbility`, can add a constructor parameter to it for `AttachmentType` and use it for this as well as Dead-Iron Sledge

Reply:
I've added the `AttachmentType` to the constructor, altered the text so that it can work with either enchanted creatures or equipped creatures.

But currently `BlocksOrBlockedAttachedTriggeredAbility` is only being used in `Ferocity` and `Gift Of The Woods` where both effects benefit the equipped creature when it is blocked or blocks, applying it to Barrow Blade makes the equipped creature lose its abilities instead of the blocking or blocked creature. 

Any suggested way to get around this? I guess some changes need to be done to the class or am I missing something?

Reply:
Ahhh that's not as straightforward as I hoped, darn. There's got to be a way for the trigger to set the target pointer to the other creature instead, might take some conditionals because checking two types of game events, distinguish which target pointer is desired with boolean in constructor I guess.

Given the additional complexity of using the common class here, I will accept the custom implementation for now, this can just be one of many potential future refactors.

Reply:
I did pull off a really janky way of getting it to work, it is not at all an optimal way of doing it lol, it only works because equipment cards will have "that creature" in the text.

```
public boolean checkTrigger(GameEvent event, Game game) {
        Permanent creature = game.getPermanent(event.getTargetId());
        Permanent equipment = game.getPermanent(sourceId);
        

        if (equipment.getAbilities().getValue().contains("that creature")){
            if (event.getSourceId().equals(equipment.getAttachedTo())) {
                Permanent blocks = game.getPermanent(event.getTargetId());
                System.out.println("Blocks = " + blocks);
                getEffects().setTargetPointer(new FixedTarget(event.getTargetId(), game));
                return true;
            }

            if (event.getTargetId().equals(equipment.getAttachedTo())) {
                Permanent blockedBy = game.getPermanent(event.getSourceId());
                System.out.println("blockedBy = " + blockedBy);
                getEffects().setTargetPointer(new FixedTarget(event.getSourceId(), game));
                return true;
            }
        }

        if (creature == null || !creature.getAttachments().contains(getSourceId())) {
            return false;
        }
        getEffects().setTargetPointer(new FixedTarget(creature, game));
        return true;
    }
```

Reply:
Anyway I'll push the other changes in a bit for you to review

Reply:
Came up with a solution using a boolean `selfTarget`

```
public boolean checkTrigger(GameEvent event, Game game) {
        Permanent creature = game.getPermanent(event.getTargetId());
        Permanent equipment = game.getPermanent(sourceId);

        if (creature == null || !creature.getAttachments().contains(getSourceId())) {
            return false;
        }

        if (!selfTarget){
            if (event.getSourceId().equals(equipment.getAttachedTo())) {
                Permanent blocks = game.getPermanent(event.getTargetId());
                getEffects().setTargetPointer(new FixedTarget(blocks, game));
            }

            if (event.getTargetId().equals(equipment.getAttachedTo())) {
                Permanent blockedBy = game.getPermanent(event.getSourceId());
                getEffects().setTargetPointer(new FixedTarget(blockedBy, game));
            }
            return true;
        }

        getEffects().setTargetPointer(new FixedTarget(creature, game));
        return true;
    }
```
Had to also change `checkEventType()`

```
return event.getType() == GameEvent.EventType.BLOCKER_DECLARED;
```

Here are some examples:

**Barrow-Blade**
```
this.addAbility(new BlocksOrBlockedAttachedTriggeredAbility(
                new LoseAllAbilitiesTargetEffect(Duration.EndOfTurn)
                .setText("that creature loses all abilities " +
                "until end of turn."), AttachmentType.EQUIPMENT, false));
```

**Ferocity**
```
this.addAbility(new BlocksOrBlockedAttachedTriggeredAbility(
                new AddCountersTargetEffect(CounterType.P1P1.createInstance())
                .setText("put a +1/+1 counter on it"),
                AttachmentType.AURA,
                true, true
        ));
```

Reply:
Oh that looks very promising now. Can you push that instead if it's ready?

Reply:
Tested and pushed it.